privacyidea/lib/crypto.py

```diff
@@ -469,6 +469,8 @@ def geturandom(length=20, hex=False):
         
     if hex:
         ret = to_unicode(binascii.hexlify(ret))
+        # 1 bytes is 2 nibbles
+        ret = ret[:length]
```

Comment:
To my understanding this must be fixed in the TiQR and not here! Please remove this.
If you do a geturandom(length=2, Hex=True) the intended behaviour is to get two random bytes in hex format! like ``AB12``.

Reply:
The length can also be an odd number, so we could pass length/2 rounded up and then truncate it to the correct length afterwards.

Reply:
OK. fine! :-)